test/unittests/test_dd_synthesis_dc.cpp

```diff
@@ -76,20 +62,49 @@ TEST_P(TestDDSynthDc, GenericDDSynthesisDcTest) {
     EXPECT_TRUE(readPla(tt, fileName));
     extend(tt);
 
-    encodeHuffman(tt);
+    TruthTable const ttOri(tt); //unchanged Truth Table.
 
-    augmentWithConstants(tt);
+    //n -> No. of primary inputs.
+    //m -> No. of primary outputs.
+    //k1 -> Minimum no. of additional lines required.
+    //codewords -> Output patterns with the respective codewords.
+    //totalNoBits -> Total no. of bits required to create the circuit
+    // r -> additional variables/bits required to decode r MSB bits of the circuit.
+
+    const auto n = tt.nInputs();
+    const auto m = tt.nOutputs();
+
+    const auto [codewords, k1] = encodeHuffman(tt);
+
+    const auto totalNoBits = std::max(n, m + k1);
+    const auto r           = totalNoBits - tt.nOutputs();
+
+    augmentWithConstants(tt, totalNoBits);
 
     const auto ttDD = buildDD(tt, dd);
     EXPECT_TRUE(ttDD.p != nullptr);
 
-    DDSynthesizer synthesizer(tt.nInputs());
-    const auto&   qc = synthesizer.synthesize(ttDD, dd);
+    DDSynthesizer synthesizer(totalNoBits);
+
+    // codewords, r, m required for decoding purposes.
+    const auto& qc = synthesizer.synthesize(ttDD, codewords, r, m, dd);
```

Comment:
It was already stated above, but this should not be in the tests in this format. This should probably be the function body of the general purpose truth table synthesis function.
In the tests, this should probably just be something like
```c++
DDSynthesizer synthesizer(tt);
const auto& qc = synthesize();
```
where the first line constructs the synthesizer and stores some pointer to the TT (probably a `std::shared_ptr`) and the second function synthesises the truth table and returns the circuit (here, it would probably also make sense to internally store the circuit `qc` as a `shared_ptr` in the synthesizer and return that pointer from the routine. In that fashion, the circuit object lives on, even if the synthesizer gets destroyed. This also opens up the possibility for a
```c++
static auto synthesize(const auto& TruthTable tt) -> std::shared_ptr<qc::QuantumComputation>;
```
method in the `DDSynthesizer` class, which could very easily be exposed to Python in a future PR.

Reply:
The static function definition is still missing.